### PR TITLE
[UPDATED]- ADDED SETUP INSTRUCTIONS TO README

### PR DESCRIPTION
**Pull request by Perelyn-sama**
- ADDED SETUP INSTRUCTIONS TO README.md
- UPDATED PRAGMA VERSION IN SOME CONTRACTS
- UPDATED FOUNDRY CONFIG LINK
- Added tests 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/Solidity/pull/63"></a>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,6 +32,32 @@ This is going to be used for our partnership with [Dev Protocol](https://devprot
...
+
+Build project 
+```bash 
+  forge build
+```
+
+Run tests 
+```bash 
+  # forge test --match-contract <contract-name>Test
+  forge test --match-contract AbsTest
```
Comment (Panquesito7) [inline README.md:58]: We should probably integrate commands for these in NPM. What do you think?

Comment (Perelyn-sama) [reply README.md:58]: I don't think that's necessary, NPM would slow things down. 

if the length of the command is the issue, we could use `forge test --mc <contract-name>Test`, or we could use a makefile like this - https://github.com/nicolasgarcia214/damn-vulnerable-defi-foundry/blob/master/makefile.

**Review by sreekar9601 — DISMISSED** [review]
Doc changes and tests look good to me!

File: README.md [hunk 2]
```diff
@@ -32,6 +32,32 @@ This is going to be used for our partnership with [Dev Protocol](https://devprot
 
 Implementations are for learning purposes only. As they may be less secure than in real-world context, use them at your discretion.
 
+## Set up 
+
+Guide to install foundry - https://book.getfoundry.sh/getting-started/installation
```
Comment (Panquesito7) [inline README.md:37]:
```suggestion
Guide to install `foundry` - <https://book.getfoundry.sh/getting-started/installation>
```

File: README.md [hunk 3]
```diff
@@ -32,6 +32,32 @@ This is going to be used for our partnership with [Dev Protocol](https://devprot
 
 Implementations are for learning purposes only. As they may be less secure than in real-world context, use them at your discretion.
 
+## Set up
```
Comment (Panquesito7) [inline README.md:35]:
```suggestion
## Setting up `foundry`
```

File: README.md [hunk 4]
```diff
@@ -32,6 +32,32 @@ This is going to be used for our partnership with [Dev Protocol](https://devprot
...
+
+Guide to install foundry - https://book.getfoundry.sh/getting-started/installation
+
+Clone project and enter directory
+```bash
+  git clone https://github.com/TheAlgorithms/Solidity.git
+  cd Solidity
+```
+
+Install forge-std(Forge Standard Library)
```
Comment (Panquesito7) [inline README.md:45]:
```suggestion
Installing `forge-std` (Forge Standard Library).

```

File: README.md [hunk 5]
```diff
@@ -32,6 +32,32 @@ This is going to be used for our partnership with [Dev Protocol](https://devprot
...
+```bash
+  forge install foundry-rs/forge-std 
+```
+
+Build project
```
Comment (Panquesito7) [inline README.md:50]:
```suggestion
Build the project using `forge-std`.

```

File: README.md [hunk 6]
```diff
@@ -32,6 +32,32 @@ This is going to be used for our partnership with [Dev Protocol](https://devprot
...
+```bash
+  forge install foundry-rs/forge-std 
+```
+
+Build project 
+```bash 
+  forge build
+```
+
+Run tests
```
Comment (Panquesito7) [inline README.md:55]:
```suggestion
Run `forge-std` automatic tests.

```

File: README.md [hunk 7]
```diff
@@ -32,6 +32,36 @@ This is going to be used for our partnership with [Dev Protocol](https://devprot
 
 Implementations are for learning purposes only. As they may be less secure than in real-world context, use them at your discretion.
 
+## Setting up `foundry`
+
+Guide to install `foundry` - <https://book.getfoundry.sh/getting-started/installation>
+
+Clone the current project and enter the directory.
```
Comment (Panquesito7) [inline README.md:39]:
```suggestion
Guide to installing `foundry` - <https://book.getfoundry.sh/getting-started/installation>\
Clone the current project and enter the directory.
```

**Review by mkubdev — APPROVED** [review]
Hello @Perelyn-sama ! :wave: Thank you so much for your contribution. LGTM ! 🚀

**Review by Panquesito7 — APPROVED** [review]
Awesome! Thanks. 🚀

**Pull request merged**